### PR TITLE
Add app to show capitals of countries

### LINKED ISSUE #99: [Proposal] Capital of <country> app
# Summary

Create an app which shows capitals of countries in the world

# Example search queries

- capital of pakistan
- india capital
- America capital
- what is capital of Sweden

# Implementation (optional)

Keep a simple CSV, JSON, YAML, JS, sheet storage which can be used. This app will demonstrate that it's possible to create knowledge apps like these with simple storage like a sheet.

- [x] Add a country capital CSV
- [x] Create a query parser for queries in format `Capital of <Country name>`
- [x] Show the result in format `Captital of <Country Name> is **<City Name>**`
- [ ] **Bonus**: Show an info card for the city
- [ ] **Bonus**: Show capitals of the states/other territories as well

# Context

### PR DESCRIPTION
**Pull request by hargup**
Screenshot 2021-09-16 at 2 51 42 PM

closes #99

### CODE REVIEW

**Comment by hargup** [comment]
Time taken: 2 hours 30 minutes.

- Things which took time, dealing with CSV was annoying. Need to bundle csv with webpack and also wanted to ensure it works on the server. Switched to JSON midway.
- Took ~30 minutes to display the result in one line. Realized, I needed to use `span` for inline styling of text instead of `div`
- Query parsing is slightly annoying, not super annoying.
- Got slighly confused with the name of the props of `Renderer`. The sample app should ideally pass some data from `queryToData` to `Renderer` for illustration.

Things we can improve:
- The app creation script should take the app name as parameter, and add an entry into the index automatically.
- Add utility function for basic kind of query parsing.

File: src/apps/capitals.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import styled from "styled-components";
+import capitalsData from "./capitalsData.json";
+
+const Territory = styled.div`
```
Comment (OrkoHunter) [inline src/apps/capitals.jsx:4]:
```suggestion
const Territory = styled.span`
```

File: src/apps/capitals.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  text-transform: capitalize;
+`;
+
+const Capital = styled.div`
```
Comment (OrkoHunter) [inline src/apps/capitals.jsx:8]:
```suggestion
const Capital = styled.span`
```

File: src/apps/capitals.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+`;
+
+//=========================================
+
+function Renderer({ data }) {
+  return (
+    <div>
+      Capital of <Territory>{data.territory}</Territory> is{" "}
+      <Capital>{data.capital}</Capital>
+    </div>
```
Comment (OrkoHunter) [inline src/apps/capitals.jsx:20]: Divs are causing unnecessary line breaks 
Screenshot 2021-09-16 at 1 26 51 PM

Screenshot 2021-09-16 at 1 27 10 PM

File: src/apps/capitals.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  // Todo @Harsh: There is probably a smarter way to do this
+  // Todo @Harsh: Replace this with a standard utility function
+  const triggerQueries = [
+    "what is the capital of",
+    "what is capital of",
+    "capital of",
+    "capital",
+  ];
+  for (const triggerQuery of triggerQueries) {
+    if (normalizedQuery.startsWith(triggerQuery)) {
```
Comment (OrkoHunter) [inline src/apps/capitals.jsx:41]: Missed opportunity for serving "Sweden capital". 😅 

Screenshot 2021-09-16 at 1 29 21 PM

File: src/apps/capitals.jsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      const TerritoryName = normalizedQuery.replace(triggerQuery, "").trim();
+      const CapitalName = territoryToCapital(TerritoryName);
+      if (CapitalName) {
+        return { territory: TerritoryName, capital: CapitalName };
```
Comment (OrkoHunter) [inline src/apps/capitals.jsx:45]: nit: Sorry! I thought these were components based on their case. :)

```suggestion
      const territoryName = normalizedQuery.replace(triggerQuery, "").trim();
      const capitalName = territoryToCapital(territoryName);
      if (capitalName) {
        return { territory: territoryName, capital: capitalName };
```

**Review by OrkoHunter — APPROVED** [review]
Since this app has two files (jsx and json), can we try creating a folder for it?

File: src/apps/capitals.jsx (hunk 4 above)
Comment (hargup) [reply src/apps/capitals.jsx:45]: aah yes, it should "includes"

File: src/apps/capitals.jsx (hunk 3 above)
Comment (hargup) [reply src/apps/capitals.jsx]: Fixed, thanks!

Actually this was in my local changes, missed pushing it 😅

File: src/apps/capitals.jsx (hunk 4 above)
Comment (hargup) [reply src/apps/capitals.jsx]: Fixed this

**Comment by hargup** [comment]
Everything looks good. Merging
Screenshot 2021-09-16 at 2 51 42 PM

**Pull request merged**